src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1833,17 +1836,23 @@ public static SqlConnection OpenSqlConnection(ConnectionInfo connInfo, string fe
                 bool? originalPersistSecurityInfo = connInfo.ConnectionDetails.PersistSecurityInfo;
                 bool? originalPooling = connInfo.ConnectionDetails.Pooling;
 
+                // allow pooling connections for language service feature to improve intellisense connection retention and performance.
+                bool shouldForceDisablePooling = featureName != Constants.LanguageServiceFeature;

```

Comment:
>bool [](http://example.com/codeflow?start=16&length=4)

`var` is your friend! Maybe other folks in ADS disagree though

Reply:
If type is known, I'd rather use it - strong typings help will nullables, var doesn't :)